clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -458,10 +462,30 @@ public KafkaProducer(Properties properties, Serializer<K> keySerializer, Seriali
         }
     }
 
-    private static TransactionManager configureTransactionState(ProducerConfig config, LogContext logContext, Logger log) {
+    private static int configureDeliveryTimeout(ProducerConfig config, Logger log) {
+        int deliveryTimeoutMs = config.getInt(ProducerConfig.DELIVERY_TIMEOUT_MS_CONFIG);
```

Comment:
Should this be of type long ?
With long, there is no overflow on line 478

In ProducerBatch, deliveryTimeoutMs is long in hasReachedDeliveryTimeout

Reply:
`ProducerBatch.hasReachedDeliveryTimeout` is called by RecordAccumulator. In RecordAccumulator's construct, we have had using `long` type for `lingerMs`, and `retryBackoffMs`. It will be inconsistent to use `int` for `deliveryTimeoutMs`. And it will require changes at many places (especially in the test cases) if we use `int` type for `lingerMs` and `retryBackoffMs`.  

        public RecordAccumulator(LogContext logContext,
                                 int batchSize,
                                 CompressionType compression,
                                 long lingerMs,
                                 long retryBackoffMs,
                                 ...